docs/misc/message_system.md

```diff
@@ -0,0 +1,242 @@
+                    */
+                    "duration": {
+                        "from": "2021-03-01T12:10:00.000Z",
+                        "to": "2022-01-31T12:10:00.000Z"
+                    },
+                    /* 
+                    For os, environment, browser and transport.
+                    - Values can be array, string or null.
+                    - Semver npm library is used for working with versions https://www.npmjs.com/package/semver.
+                    - "*" = all versions; "!" = not for this type of browser/os/...
+                    - Options: gte, lt, ranges, tildes, carets,... are supported, see semver lib for more info.
+                    */
+                    "os": {
+                        "macos": [
+                            "10.14",
+                            "10.18",
+                            "11"
+                        ],
+                        "linux": "<20.04",
+                        "windows": "!",
+                        "android": "*",
+                        "ios": "13"
+                    },
+                    "environment": {
+                        "desktop": "<21.5",
+                        "mobile": "!",
+                        "web": "<22"
+                    },
+                    "browser": {
+                        "firefox": [
+                            "82",
+                            "83"
+                        ],
+                        "chrome": "*",
+                        "chromium": "!"
+                    },
+                    "transport": {
+                        "bridge": [
+                            "2.0.30",
+                            "2.0.27"
+                        ],
+                        "webusbplugin": "*"
+                    },
+                    /*
+                    - If key is not available (undefined), then it can be whatever.
+                    - Currently supported keys are "tor" and coin symbols from "enabledNetworks".
+                    - The bitwise operation is OR.
+                    */
+                    "settings": [
+                        { 
+                            "tor": true,
+                            "btc": true
+                        },
+                        {
+                            "tor": false,
+                            "ltc": true
+                        }
+                    ],
+                    // Empty device array is targeting users without a connected device.
+                    "devices": [ 
+                        {
+                            // Possible values: "1" or "T"
+                            "model": "1", 
+                            "firmware": "1.9.4",
+                            "vendor": "trezor.io"
+                        }
+                    ]
+                }
+            ],
+            "message": {
+                // Used for remembering dismissed messages.
+                "id": "0f3ec64d-c3e4-4787-8106-162f3ac14c34",
```

Comment:
Is this a random UUID or should it match something?

Reply:
Yes, random UUID. It has to be unique to avoid displaying dismissed messages twice.